service/frontend/workflow_handler.go

```diff
@@ -1696,13 +1711,13 @@ func (wh *WorkflowHandler) SignalWithStartWorkflowExecution(ctx context.Context,
 		return nil, err
 	}
 
-	request, err := wh.unaliasSignalWithStartWorkflowExecutionRequestSearchAttributes(request, namespaceName)
+	sa, err := wh.unaliasedSearchAttributesFrom(request.GetSearchAttributes(), namespaceName)
 	if err != nil {
 		return nil, err
 	}
-
-	if err = wh.validateSearchAttributes(request.GetSearchAttributes(), namespaceName); err != nil {
-		return nil, err
+	if sa != request.GetSearchAttributes() {
+		request = common.CloneProto(request)
+		request.SearchAttributes = sa
```

Comment:
why not `newRequest := common.CloneProto(request)`?

separate nitpick, I liked how the deleted functions called the new search attributes `unaliasedSas` instead of just `sa` like you have. Imo it's slightly easier to follow what's happening with a more descriptive variable name than just `sa`

Reply:
ohh I see now that using `request` instead of `newRequest` means less modification to the rest of the code that refers to `request` and expects the new unaliased request